hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/ECKeyOutputStream.java

```diff
@@ -118,7 +118,7 @@ public long getFlushCheckpoint() {
 
   private ECKeyOutputStream(Builder builder) {
     super(builder.getReplicationConfig(), builder.getClientMetrics());
-    this.config = builder.getClientConfig();
+    this.config = builder.getClientConfig().cloneSelf();
```

Comment:
`clientConfig` is passed to the builder by `RpcClient` here:

https://github.com/apache/ozone/blob/8202b4184e57dede01f21bc5be5563ae85d1dc5b/hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java#L2257

It should create a new object using the existing method:

```java
        .setConfig(conf.getObject(OzoneClientConfig.class))
```

And apply the same fix in other places where `clientConfig` is passed similarly:

https://github.com/apache/ozone/blob/8202b4184e57dede01f21bc5be5563ae85d1dc5b/hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java#L1796

https://github.com/apache/ozone/blob/8202b4184e57dede01f21bc5be5563ae85d1dc5b/hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java#L2187

Reply:
Yes, this is better. @adoroszlai Thanks.